tests/unit/transformer/test_mapper.py

```diff
@@ -299,3 +309,87 @@ def test_ambiguous_column_names_exception_is_ignored(self, input_df, mapping, ex
+        transformer = Mapper(matching_mapping)
+        assert_df_equality(input_df, transformer.transform(input_df))
+
+    def test_matching_mapping_with_validation(self, input_df, matching_mapping):
+        transformer = Mapper(matching_mapping, mode="rename_and_validate")
+        assert_df_equality(input_df, transformer.transform(input_df))
+
+    def test_mismatching_mapping_with_casting(self, input_df, matching_mapping):
+        mapping_ = deepcopy(matching_mapping)
+        mapping_[0] = ("col_a", "col_a", T.StringType())
+        transformer = Mapper(mapping_)
+        mapped_df = transformer.transform(input_df)
+        assert input_df.columns == mapped_df.columns
+        assert mapped_df.schema["col_a"].jsonValue()["type"] == "string"
+
+    def test_mismatching_mapping_with_validation(self, input_df, matching_mapping):
+        mapping_ = deepcopy(matching_mapping)
+        mapping_[0] = ("col_a", "col_a", T.StringType())
+        transformer = Mapper(mapping_, mode="rename_and_validate")
+        with pytest.raises(DataTypeValidationFailed, match="col_a"):
+            transformer.transform(input_df)
+
+    def test_spooq_transformation_raises_exception(self, input_df, matching_mapping):
+        mapping_ = deepcopy(matching_mapping)
+        mapping_[0] = ("col_a", "col_a", spq.to_int)
+        transformer = Mapper(mapping_, mode="rename_and_validate")
+        with pytest.raises(DataTypeValidationFailed, match="Spooq transformations are not allowed in 'rename_and_validate' mode"):
+            transformer.transform(input_df)
+
+    def test_mapper_raises_exception_for_missing_column(self, input_df, matching_mapping):
+        missing_column = "col_d"
+        mapping_ = deepcopy(matching_mapping)
+        mapping_.append((missing_column, missing_column, T.StringType()))
+        transformer = Mapper(mapping_, mode="rename_and_validate", missing_column_handling="raise_error")
+        with pytest.raises(AnalysisException, match=f"A column or function parameter with name `{missing_column}` cannot be resolved."):
+            transformer.transform(input_df)
+
+    def test_validation_raises_exception_for_skipped_column(self, input_df, matching_mapping):
+        skipped_column = "col_d"
+        mapping_ = deepcopy(matching_mapping)
+        mapping_.append((skipped_column, skipped_column, T.StringType()))
+        transformer = Mapper(mapping_, mode="rename_and_validate", missing_column_handling="skip")
+        with pytest.raises(DataTypeValidationFailed, match=f"Input column: {skipped_column} not found!"):
+            transformer.transform(input_df)
+
+    def test_validation_succeeds_for_nullified_column(self, input_df, matching_mapping):
+        column_to_nullify = "col_d"
+        mapping_ = deepcopy(matching_mapping)
+        mapping_.append((column_to_nullify, column_to_nullify, T.StringType()))
+        mapped_df = Mapper(mapping_, mode="rename_and_validate", missing_column_handling="nullify").transform(input_df)
```

Comment:
did you think about making the mode an enum, like:
```suggestion
        mapped_df = Mapper(mapping_, mode=MapperMode.RENAME_AND_VALIDATE, missing_column_handling="nullify").transform(input_df)
```

same as above, no need to implement this now